### PR TITLE
fix publish fail on prexisting bad record

### PR DESCRIPTION
**Pull request by whyrusleeping**
dont error out if prexisting record is bad, just grab its sequence number
and continue on with the publish.

License: MIT
Signed-off-by: Jeromy <jeromyj@gmail.com>

### CODE REVIEW

**Comment by whyrusleeping** [comment]
tested on four different nodes live on the network, mars, my desktop, my laptop, and one other machine i've had online forever, publishing and resolution works across the board there, even when updating from v0.3.3 to latest on a node that previously had an ipns entry.

I want to use something (like nomad) for staging and testing network upgrades for things like this. I'm thinking of having a docker image 'ipfs-bot' that will start ipfs, connect either to the primary network, or just bootstrap to our other test nodes (leaning towards the main network), and then run random commands and sets of operations.

**Comment by jbenet** [comment]
@whyrusleeping re test network, this is what i want with golang/build. i'm looking more at nomad and it seems tuned much more for stationary things, not for running random, isolated, resource constrained, one-off tasks. it may be that we can use nomad to orchestrate managing the builder processes, but running one-off taksa (builds or tests), the dashboard, viewable logs, etc, don't seem to be covered by nomad.

**Comment by whyrusleeping** [comment]
I think we're talking about different things here. I want to have a network of long lived nodes, and have the ability to say 'update 20% of our test nodes to this new builds, and have them run a bunch of tasks', or 'wipe half of our test network, bring that many new nodes online, and see what happens'

File: namesys/publisher.go [hunk 1]
```diff
@@ -62,22 +67,44 @@ func (p *ipnsPublisher) PublishWithEOL(ctx context.Context, k ci.PrivKey, value
...
 		if err != nil {
-			return err
+			return 0, err
 		}
 
-		seqnum = e.GetSequence() + 1
+		return e.GetSequence(), nil
 	} else if err != ds.ErrNotFound {
-		return err
+		return 0, err
```
Comment (jbenet) [inline namesys/publisher.go]: this won't work with a node whose local repo is destroyed, but has records published. you'll end up with records at very low seqnos that won't overwrite the others. two options: 

- request the seqnos
- use the creation timestamp as a seqno (so, two timestamps, one for creation, one for publication)

Comment (jbenet) [inline namesys/publisher.go]: this won't work with a node whose local repo is destroyed, but has records published. you'll end up with records at very low seqnos that won't overwrite the others. two options:

request the record, see what comes back, take the max and go from there
use the creation timestamp as a seqno (so, two timestamps, one for creation, one for publication)

Comment (whyrusleeping) [inline namesys/publisher.go]: hrm... okay.

**Comment by whyrusleeping** [comment]
@jbenet alright. I think weve got this one in the bag.

**Comment by whyrusleeping** [comment]
like, more in the bag than this cat: 
image

**Comment by jbenet** [comment]
LGTM!

**Comment by jbenet** [comment]
that's a great cat

**Pull request merged**